### PR TITLE
Adds support for non-wrapping iso rendering

### PR DESCRIPTION
**Pull request by ninjasort**
Given a React component contains a `iso-root` class on it's html element along with a data-key attribute and a script tag with a comment (`<!-- ___iso-state___ -->`), this PR allows for a truly isomorphic render to be performed from the root of the document resulting in a minimal payload to the client without server templating. 
### Example:

_App.js_

```
import React from 'react';
import Router, {RouteHandler} from 'react-router';

export default class App extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <html className="iso-root" data-key>
        <head>
          <meta charset="utf-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no" />
          <meta name="description" content="" />
          <title>{this.props.title}</title>
          <link rel="stylesheet" href="/css/core.css" />
        </head>
        <body>
          <RouteHandler {...this.props} />
          <script src="/js/bundle.js"></script>
          <span dangerouslySetInnerHTML={{__html: '<!--___iso-state___-->'}} />
        </body>
      </html>
    );
  }
}
```

_server.js_

```
...

app.use(function(req, res) {
  alt.bootstrap(JSON.stringify(res.locals || {}));

  var iso = new Iso();

  const router = Router.create({
    routes: routes,
    location: req.url
  });

  router.run(function (Handler, state) {
    var html = React.renderToString(React.createElement(Handler));
    iso.add(html, alt.flush());

    res.send('<!DOCTYPE html>' + iso.render());
  });
});
```

### CODE REVIEW

**Comment by jfsiii** [comment]
I think state should be in a `script` tag; not a `span`.

This is [what @satyam suggested](https://github.com/goatslacker/iso/issues/9#issuecomment-124566221) in the original thread, and @goatslacker [agreed](https://github.com/goatslacker/iso/issues/9#issuecomment-132082075)

**Comment by ninjasort** [comment]
I've added a script tag with the state and it's bootstrapping on the client, any idea why the react-router example isn't working correctly?

File: package.json [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
   "scripts": {
+    "test": "mocha -w --compilers js:babel/register",
     "build": "babel ./src/iso.js > dist/iso.js; uglifyjs dist/iso.js -m > dist/iso.min.js"
   },
   "devDependencies": {
     "babel": "5.8.21",
+    "babelify": "^6.2.0",
+    "browserify": "^11.0.1",
+    "chai": "^3.2.0",
+    "jasmine-core": "^2.3.4",
```
Comment (michaelBenin) [inline package.json:17]: What is jasmine-core needed for?

**Comment by goatslacker** [comment]
Ooo nice and with tests. Will review this.

File: package.json (hunk 1 above)
Comment (ninjasort) [inline package.json:17]: It's a dependency for running jasmine with karma.

Comment (michaelBenin) [inline package.json:17]: Are you sure it's being used? In the karma config jasmine isn't referenced.

Comment (ninjasort) [inline package.json:17]: I don't believe so actually, but I didn't install it manually.

**Comment by michaelBenin** [comment]
Any news on this @goatslacker?

**Pull request closed without merging**